include/ignition/rendering/ShaderParam.hh

```diff
@@ -64,29 +72,50 @@ namespace ignition
       /// \return Type of this parameter
       public: ParamType Type() const;
 
-      /// \brief Set from another ShaderParam.
-      /// \param[in] _other Another ShaderParam.
-      /// \return Reference to this ShaderParam.
+      /// \brief Get the element count of this parameter's buffer
+      /// \return Count of elements in this parameter's buffer
+      public: uint32_t Count() const;
+
+      /// \brief Set from another ShaderParam
+      /// \param[in] _other Another ShaderParam
+      /// \return Reference to this ShaderParam
       public: ShaderParam &operator=(const ShaderParam &_other);
 
-      /// \brief Set this to be a float param
-      /// \param[in] _value Value to set this param to.
+      /// \brief Set this to be a float parameter
+      /// \param[in] _value Value to set this parameter to
       public: void operator=(const float _value);
 
-      /// \brief Set this to be an integer param;
-      /// \param[in] _value Value to set this param to.
+      /// \brief Set this to be an integer parameter
+      /// \param[in] _value Value to set this parameter to
       public: void operator=(const int _value);
 
+      /// \brief Set this to be a word buffer parameter
+      /// \param[in] _count Number of elements in the word buffer
+      public: void InitializeWordBuffer(const uint32_t _count);
+
+      /// \brief Copy a buffer to this parameter
+      /// \param[in] _floatBuffer Source buffer to copy from
+      public: void UpdateWordBuffer(float *_floatBuffer);
```

Comment:
how about just `InitBuffer` and `UpdateBuffer` to keep the names concise? unless `Word` has a semantic meaning?

Reply:
The `InitWordBuffer` and `UpdateWordBuffer` only work for 32bit data types. There is a extension _ARB_gpu_shader_int64_ to support 64bit data types but I am not too sure if ogre is going to implement it.

Since the buffer is generic `std::shared_ptr<void>` it can support any size data types, but the methods that use the buffer use element (4 bytes) count rather than buffer size which might cause problems when trying to create a buffer of 64bit data types.

I am not to sure how often 64bit data types are used for uniforms so I could probably just change it to `InitBuffer` and `UpdateBuffer` but add a comment saying only 32bit data type uniforms are supported.

Reply:
> I am not to sure how often 64bit data types are used for uniforms so I could probably just change it to InitBuffer and UpdateBuffer but add a comment saying only 32bit data type uniforms are supported.

sounds good. We've only been using 32bit data type uniforms in our shaders and I think there are also other places that make this implicit assumption. I'm ok with adding doc to let users know that only 32bit types are supported.